Distributions.ecl

```diff
@@ -0,0 +1,407 @@
+    #option pure;
+    double work_sum = (df%2==0) ? 1.0  : c;
+    double work_term = (df%2==0) ? 1.0  : c;
+    double start = (df%2==0) ? 2.0  : 3.0;
+    double stop = (double) (df - 2);
+    for (double ai=start; ai<=stop; ai+=2.0) {
+      work_term *= csq * ((ai-1.0)/ai);
+      work_sum  += work_term;
+    }
+    return work_sum;
+  ENDC++;
+  /**
+   * Students t distribution integral evaluated between negative
+   * infinity and x.
+   * Translated from NIST SEL DATAPAC Fortran TCDF.f source
+   * @param x value of the evaluation
+   * @param df degrees of freedom
+   * @returns the probability that a value will be less than the
+   * specified value
+   */
+  EXPORT REAL8 T_CDF(REAL8 x, REAL8 df) := FUNCTION
+    INTEGER4 DF_Cut := 1000;
+    REAL8 PI := Core_Const.Pi;
+    INTEGER4 ndf := (INTEGER4) df;
+    BOOLEAN isEven := ndf%2 = 0;
+    // Small and moderate df definitions
+    REAL8 sd := SQRT(df/(df-2.0));
+    REAL8 z := x/sd;
+    REAL8 c := SQRT(df/(x*x+df));
+    REAL8 csq := df/(x*x+df);
+    REAL8 s := x/SQRT(x*x+df);
+    REAL8 sum_terms := MAP(
+        df=1  => (2/PI)*(ATAN(x)),
+        df=2  => s,
+        df=3  => (2/PI)*(ATAN(x/SQRT(df))+(c*s)),
+        isEven=> Sum_Terms(df, c, csq) * s,
+        (2/PI)*(ATAN(x/SQRT(df)) + Sum_Terms(df, c, csq)*s));
+    // Large df asymptoptic approximation definitions
+    REAL8 DCONST := 0.3989422804;
+    REAL8 B11 := 0.25;
+    REAL8 B21 := 0.01041666666667;
+    REAL8 B22 := 3.0;
+    REAL8 B23 := -7.0;
+    REAL8 B24 := -5.0;
+    REAL8 B25 := -3.0;
+    REAL8 B31 := 0.00260416666667;
+    REAL8 B32 := 1.0;
+    REAL8 B33 := -11.0;
+    REAL8 B34 := 14.0;
+    REAL8 B35 := 6.0;
+    REAL8 B36 := -3.0;
+    REAL8 B37 := -15.0;
+    REAL8 d1 := x;
+    REAL8 d3 := d1 * x * x;
+    REAL8 d5 := d3 * x * x;
+    REAL8 d7 := d5 * x * x;
+    REAL8 d9 := d7 * x * x;
+    REAL8 d11:= d9 * x * x;
+    REAL8 df_2 := df * df;
+    REAL8 df_3 := df_2 * df;
+    REAL8 Term1 := B11*(d3+d1)/df;
+    REAL8 Term2 := B21*(B22*d7+B23*d5+B24*d3+B25*d1)/df_2;
+    REAL8 Term3 := B31*(B32*d11+B33*d9+B34*d7+B35*d5+B36*d3+B37*d1)/df_3;
+    REAL8 Norm_CDF := Normal_CDF(x);
+    T_CDF := MAP(df <= 0                             => 0.0,
+                df BETWEEN 3 AND 9     AND z<= -3000 => 0.0,
+                df BETWEEN 3 AND 9     AND z>=  3000 => 1.0,
+                df BETWEEN 10 AND 1000 AND z<= -150  => 0.0,
+                df BETWEEN 10 AND 1000 AND z>=  150  => 1.0,
+                df BETWEEN 1 AND 1000                => 0.5+sum_terms/2.0,
+                Norm_CDF-(EXP(-x*x*0.5))*(Term1+Term2+Term3)*DCONST);
+    RETURN T_CDF;
+  END;
+
+  // Inverse T helper functions
+  REAL8 df3_helper(REAL8 trm, REAL8 init_z) := BEGINC++
+    #option pure;
+    #include <math.h>
+    double c = cos(init_z);
+    double s = sin(init_z);
+    double z = init_z;
+    for (int i=0; i<15; i++) {
+      z = z-(z+s*c-trm)/(2.0*c*c);
+      c = cos(z);
+      s = sin(z);
+    }
+    return sqrt(3)*s/c;
```

Comment:
may be worth extracting sqrt(3) as a constant:

    const static double sqrt3 = sqrt(3);
    #body
    ...
    return sqrt3 *s/c